api/submit_result.go

```diff
@@ -32,12 +36,32 @@ func (s *resultSubmitter) Submit(executionID string, outputKey string, outputDat
 	if err != nil {
 		return err
```

Comment:
We can still wrap this error in the execution. This will happen when we delete a service that is running an execution. The execution will fail but we might for example have a restart policy and when we reinstall the service it will be completed or this kind of behaviors that are possible if we store this error

Reply:
I cannot follow what you mean exactly. Do you mean calling _Failed_ by "wrapping" This error here is raised when _FromService_ is unsuccessful which it would be an internal error of core if it's not nil. I also checked the implementation of _FromService_ and it always returns nil errors for now.

Reply:
I guess @antho1404 is saying:
```
if err != nil {
	exec.Fail(err)
	return exec, nil
}
```

Reply:
It's to change the status of the exec to fail and publish it anyway.

Reply:
>  We should not fail executions on internal errors because service may re-try. I also checked the implementation of FromService and it always returns nil errors for now.

Reply:
The logic @antho1404 want to push is that's not a service's error.
The service did it job, it executed the task and provide a result.
Any errors that happen on the core should not be returned to the service but should be return the "user" by changing the state of the execution to failed.

Reply:
now, this makes sense 👍